core/domain/exp_domain.py

```diff
@@ -2460,6 +2513,48 @@ def get_all_html_content_strings(self):
 
         return html_list
 
+    @classmethod
+    def to_proto(
+            cls, exploration_id, title, version, init_state_name, states):
+        """Calculate the exploration size by setting exploration proto.
+
+        Args:
+            exploration_id: str. The id of the exploration.
+            title: str. The exploration title.
+            version: int. The version of the exploration.
+            init_state_name: str|None. The name of the initial state.
+            states: list|None. The list of state
+                domain objects.
+
+        Returns:
+            Proto Object. The exploration proto object.
+        """
+        state_protos = {}
+        exploration_proto = {}
+        if states is not None:
+            state_protos = state_domain.State.to_proto(states)
+
+        if isinstance(title, str):
+            exploration_proto = exploration_pb2.Exploration(
+                id=exploration_id,
+                content_version=version,
+                init_state_name=init_state_name,
+                title=title,
+                states=state_protos)
+        return exploration_proto
+
+    @classmethod
+    def calculate_size_of_proto(cls, proto):
```

Comment:
Why not have this be self.get_proto_size() -- which calls self.to_proto() internally and returns its size? (Do you have cases where you need to construct the proto once and then get its size multiple times?)

Reply:
Done